src/coreclr/jit/lsrabuild.cpp

```diff
@@ -1585,25 +1609,36 @@ void LinearScan::buildUpperVectorRestoreRefPosition(
 {
     if (lclVarInterval->isPartiallySpilled)
     {
-        unsigned     varIndex            = lclVarInterval->getVarIndex(compiler);
-        Interval*    upperVectorInterval = getUpperVectorInterval(varIndex);
-        RefPosition* savePos             = upperVectorInterval->recentRefPosition;
+        lclVarInterval->isPartiallySpilled = false;
+        unsigned     varIndex              = lclVarInterval->getVarIndex(compiler);
+        Interval*    upperVectorInterval   = getUpperVectorInterval(varIndex);
+        RefPosition* savePos               = upperVectorInterval->recentRefPosition;
+        if (!isUse && !savePos->liveVarUpperSave)
+        {
+            // If we are just restoring upper vector at the block boundary and if this is not
+            // a upperVector related to the liveVar, then ignore creating restore for them.
+            // During allocation, we will detect that this was an extra save-upper and skip
+            // the save/restore altogether.
+            return;
+        }
```

Comment:
What is this case about? How do we know this is at a block boundary here?

Reply:
We pass `isUse = false` at the block boundary. If they were not live variables i.e. `!liveVarUpperSave`, we know that these are the extra ones that we added and are not really needed. Had they were needed, we would have already built `RefUpperVectorRestore` at their use before we reached the block boundary.